docs/user-guide/concepts/tools/python-tools.md

```diff
@@ -281,3 +283,99 @@ When using the [`@tool`](../../../api-reference/tools.md#strands.tools.decorator
 1. If you return a string or other simple value, it's wrapped as `{"text": str(result)}`
 2. If you return a dictionary with the proper [`ToolResult`](../../../api-reference/types.md#strands.types.tools.ToolResult) structure, it's used directly
 3. If an exception occurs, it's converted to an error response
+
+## Class-Based Tools
+
+Class-based tools allow you to create tools that maintain state and leverage object-oriented programming patterns. This approach is useful when your tools need to share resources, maintain context between invocations, or follow object-oriented design principles.
+
+### Basic Example
+
+Here's how to create a class-based tool using the [`@tool`](../../../api-reference/tools.md#strands.tools.decorator.tool) decorator:
+
+```python
+from strands import Agent, tool
+
+class Browser:
+    """Browser tool implementation using Playwright."""
+
+    def __init__(self):
+        self._started = False
+        self._sessions = {}
+        # Initialize other class-level resources or state
+
+    @tool
+    def browser(self, url: str, action: str = "navigate") -> str:
+        """Interact with a web browser.
+        
+        Args:
+            url: The URL to navigate to or interact with
+            action: The action to perform (navigate, click, etc.)
+        """
+        # Method implementation that can access class instance variables
+        # and maintain state between invocations
+        return f"Performed {action} on {url}"
+
+# Usage
+browser_instance = Browser()
+agent = Agent(
+    tools=[browser_instance.browser]  # Pass the bound method as a tool
+)
+```
+
+When you use the [`@tool`](../../../api-reference/tools.md#strands.tools.decorator.tool) decorator on a class method, the method becomes bound to the class instance when instantiated. This means the tool function has access to the instance's attributes and can maintain state between invocations.
+
+### Advantages of Class-Based Tools
+
+Class-based tools offer several advantages:
```

Comment:
This is a re-stating of the introduction.  IMHO it'd be better to remove it or move this into the introduction, but as-is it's just extra fluff.